[esx]/es_extended/common/functions.lua

```diff
@@ -1,100 +1,105 @@
 local Charset = {}
 
-for i = 48,  57 do table.insert(Charset, string.char(i)) end
-for i = 65,  90 do table.insert(Charset, string.char(i)) end
-for i = 97, 122 do table.insert(Charset, string.char(i)) end
+for i = 48, 57 do ESX.TableInsert(Charset, string.char(i)) end
+for i = 65, 90 do ESX.TableInsert(Charset, string.char(i)) end
+for i = 97, 122 do ESX.TableInsert(Charset, string.char(i)) end
 
 function ESX.GetRandomString(length)
-	math.randomseed(GetGameTimer())
+    math.randomseed(GetGameTimer())
 
-	if length > 0 then
-		return ESX.GetRandomString(length - 1) .. Charset[math.random(1, #Charset)]
-	else
-		return ''
-	end
+    if length > 0 then
+        return ESX.GetRandomString(length - 1) .. Charset[math.random(1, #Charset)]
+    else
+        return ''
+    end
 end
 
 function ESX.GetConfig()
-	return Config
+    return Config
 end
 
 function ESX.GetWeapon(weaponName)
-	weaponName = string.upper(weaponName)
+    weaponName = string.upper(weaponName)
 
-	for k,v in ipairs(Config.Weapons) do
-		if v.name == weaponName then
-			return k, v
-		end
-	end
+    for k, v in ipairs(Config.Weapons) do
+        if v.name == weaponName then
+            return k, v
+        end
+    end
 end
 
 function ESX.GetWeaponFromHash(weaponHash)
-	for k,v in ipairs(Config.Weapons) do
-		if joaat(v.name) == weaponHash then
-			return v
-		end
-	end
+    for k, v in ipairs(Config.Weapons) do
+        if joaat(v.name) == weaponHash then
+            return v
+        end
+    end
 end
 
 function ESX.GetWeaponList()
-	return Config.Weapons
+    return Config.Weapons
 end
 
 function ESX.GetWeaponLabel(weaponName)
-	weaponName = string.upper(weaponName)
+    weaponName = string.upper(weaponName)
 
-	for k,v in ipairs(Config.Weapons) do
-		if v.name == weaponName then
-			return v.label
-		end
-	end
+    for k, v in ipairs(Config.Weapons) do
+        if v.name == weaponName then
+            return v.label
+        end
+    end
 end
 
 function ESX.GetWeaponComponent(weaponName, weaponComponent)
-	weaponName = string.upper(weaponName)
-	local weapons = Config.Weapons
-
-	for k,v in ipairs(Config.Weapons) do
-		if v.name == weaponName then
-			for k2,v2 in ipairs(v.components) do
-				if v2.name == weaponComponent then
-					return v2
-				end
-			end
-		end
-	end
+    weaponName = string.upper(weaponName)
+    local weapons = Config.Weapons
+
+    for k, v in ipairs(Config.Weapons) do
+        if v.name == weaponName then
+            for k2, v2 in ipairs(v.components) do
+                if v2.name == weaponComponent then
+                    return v2
+                end
+            end
+        end
+    end
+end
+
+function ESX.TableInsert(table, value)
+    table[#table + 1] = value
```

Comment:
Pretty sure this completely removes the performance benefit lol

Reply:
also, ESX.Table.Insert, would be correct naming, since ESX has a namespace for tables :)

Reply:
> Pretty sure this completely removes the performance benefit lol

Indexing the last key and inserting +1 to it as well as it's value would be indeed quicker 🙂

Reply:
> also, ESX.Table.Insert, would be correct naming, since ESX has a namespace for tables :)

This could be easily fixed...

Reply:
> > Pretty sure this completely removes the performance benefit lol
> 
> Indexing the last key and inserting +1 to it as well as it's value would be indeed quicker 🙂
 
then why not just use table.insert? it uses the current method because its faster, not because its nicer :)

Reply:
I never mentioned that it was prettier, but this is definitely faster. 👍🏼

Reply:
You did the same thing here [(977dba4)](https://github.com/esx-framework/esx-legacy/pull/618/commits/977dba4ee3532a404448825fb226a573bc2e26c8), except my commit was meant to make it a global function.

Also, if `ESX.Table.Insert` is the correct name, then `ESX.DumpTable` would also have to be renamed to `ESX.Table.Dump` since we're working with tables/arrays here.

Reply:
> Also, if ESX.Table.Insert is the correct name, then ESX.DumpTable would also have to be renamed to ESX.Table.Dump since we're working with tables/arrays here.

it should be, but that was added 4 years ago, so we cant change it lol